packit_service/worker/testing_farm.py

```diff
@@ -143,10 +143,29 @@ def _payload(
         if self.fmf_ref:
             fmf["ref"] = self.fmf_ref
 
+        predefined_environment = {
+            "full_repo_name": self.project.full_repo_name,
+            "upstream_name": self.job_config.upstream_package_name,
+            "upstream_url": self.job_config.upstream_project_url,
+            "downstream_name": self.job_config.downstream_package_name,
+            "downstream_url": self.job_config.downstream_project_url
+            if self.job_config.downstream_package_name
+            else None,
+            "commit_sha": self.metadata.commit_sha,
+        }
+        predefined_environment = {
+            k: v for k, v in predefined_environment.items() if v is not None
+        }
+        # User-defined variables have priority
+        metadata = self.job_config.metadata
+        env_variables = metadata.environment if hasattr(metadata, "environment") else {}
+        predefined_environment.update(env_variables)
```

Comment:
I am not sure if overriding is a feature or can cause some problems..;)
(But I can't think of any issue with this for now.)

Reply:
I was wondering about the same thing. The problem is that you somehow have to merge it, one way or the other. I feel like that if a user sets a variable in their config they expect it there so I'd rather not override it even if it clashes with our pre-defined variable. Your other suggestion for naming (capitalizing + prefixing `PACKIT_`) should almost completely remove the possibility of this hapening though :)